microprofile/WFLY-10712_microprofile_metrics.asciidoc

```diff
@@ -0,0 +1,304 @@
+# TYPE base:cpu_system_load_average gauge
+base:cpu_system_load_average 1.9658203125
+...
+----
+
+If security is disabled in the subsystem configuration, the HTTP client does not require authentication (and it does not require a WildFly management user either).
+
+Note that the _standalone profiles will explicitly disable authentication_ as the HTTP management interface is bound to `127.0.0.1`.
+
+==== Exposed Metrics
+
+===== Base Metrics
+
+By default, the subsystem will expose all required base metrics specified in the chapter 4 of the MicroProfile Metrics specification.
+All these base metrics are gathered from the JVM MBeans and uses smallrye-metrics JmxRegistrar to bridge from JMX to the MicroProfile Metrics API.
+The required base metrics are explicitly specified in a property file (named `base-metrics.properties`) using a set of properties for each metric:
+
+----
+<metric name>.displayName: <Human-readable name of the metric>
+<metric name>.type: <Type of metric enumerated in org.eclipse.microprofile.metrics.MetricType (e.g counter, gauge)>
+<metric name>.unit: <Unit of the metric, can be none, listed in org.eclipse.microprofile.metrics.MetricUnits or other units>
+<metric name.description: <Human-readable description of the metric>
+<metric name>.mbean: <ObjectName of the MBean and attribute>
+----
+
+For example, the properties to expose the _Total Loaded Class Count_ of the JVM are:
+
+----
+classloader.totalLoadedClass.count.displayName: Total Loaded Class Count
+classloader.totalLoadedClass.count.type: counter
+classloader.totalLoadedClass.count.unit: none
+classloader.totalLoadedClass.count.description: Displays the total number of classes that have been loaded since the Java virtual machine has started execution.
+classloader.totalLoadedClass.count.mbean: java.lang:type=ClassLoading/TotalLoadedClassCount
+----
+
+The name of the metric itself is `classloader.totalLoadedClass.count`.
+
+The list of required base metrics required to pass the MicroProfile Metrics TCK is listed at https://github.com/thorntail/thorntail/blob/master/fractions/microprofile/microprofile-metrics/src/main/resources/io/smallrye/metrics/base-metrics.properties
+
+==== Vendor Metrics
+
+Vendor metrics are specific to a "vendor" (in our case WildFly) and exposes metrics specific to the vendor runtime.
+
+Examples of such metrics are:
+
+* Number of modules loaded by JBoss Module
+* Transaction statistics from Narayana Transaction Manager
+* Bytes throughput from Undertow
+* Etc.
+
+Some of these metrics can be obtained by JMX and can rely on smallrye-metrics that loads these metrics from a property file named `vendor-metrics.properties`
+that works similarly to the `base-metrics.properties` as explained in the section above.
```

Comment:
Same question my one above about base-metrics.properties.

Overall we need to work through the requirements re: configuration.

Reply:
I've updated the spec to mention the names of the metrics exposed by the subsystem in the base and vendor spec.